### PR TITLE
added touch support

### LINKED ISSUE #34: Add multi-touch support on Android
Currently, you can only get the primary touch position using mouse_position().
However, most smartphones support multiple touch positions, and this feature is a must in most games.
As this feature already exists in microquad, it shouldn't be too hard to implement it in macroquad.

### PR DESCRIPTION
**Pull request by brianwp3000**
Resolves #34 
Added multitouch support from miniquad. I tried several different ways to structure the API, and landed on `touches() -> Vec<Touch>` because it was simple, didn't have any borrowing/lifetimes, and contained all the information needed to use touchscreen input (including the ability to track touch ids for manual gesture inputs). 

Also, the default behavior in miniquad is to treat touch events like mouse events (i.e. whenever a touch event is fired, it forwards it as a mouse event). I preserved this behavior, but added a toggle to turn it off.

I tested this by running the `input_touch.rs` example in a browser--turns out touch events aren't raised in Windows desktop apps, even if you have a touchscreen. I'm guessing that's a feature that would have to be added to miniquad.

### CODE REVIEW

**Comment by brianwp3000** [comment]
Also, I didn't know if my `Touch` struct needed the NanoSerde attribute or not. Please advise

File: src/input.rs [hunk 1]
```diff
@@ -75,3 +128,8 @@ pub fn is_mouse_button_released(btn: MouseButton) -> bool {
 
     context.mouse_released.contains(&btn)
 }
+
+/// Convert a position in pixels to a position in the range [-1; 1].
+fn convert_to_local(pixel_pos: Vec2) -> Vec2 {
+    Vec2::new(pixel_pos.x / screen_width(), pixel_pos.y / screen_height()) * 2.0 - Vec2::new(1.0, 1.0)
+}
```
Comment (optozorax) [inline src/input.rs:135]: You could run `rustfmt` on changed files

**Review by optozorax — APPROVED** [review]

**Comment by not-fl3** [comment]
Thanks for the PR!

I am ready to merge it, but could you run `rustfmt` on your changes? :) 

> Also, I didn't know if my Touch struct needed the NanoSerde attribute or not. Please advise
.
This is fine without NanoSerde, we generally do not use NanoSerde for macroquad's structs.

File: src/input.rs (hunk 1 above)
Comment (brianwp3000) [reply src/input.rs]: done

**Comment by not-fl3** [comment]
Awesome, thanks again! :)

**Pull request merged**